pkg/cloudprovider/providers/azure/azure_client.go

```diff
@@ -181,7 +209,11 @@ func (az *azVirtualMachinesClient) Get(resourceGroupName string, VMName string,
 }
 
 func (az *azVirtualMachinesClient) List(resourceGroupName string) (result compute.VirtualMachineListResult, err error) {
-	az.rateLimiter.Accept()
+	err = createArmRateLimitErr(false, "VMList")
```

Comment:
Could we optimize by putting these `err` constructor invocations inside the failure condition if block (as they are only used if the failure condition is met)?